### PR TITLE
feat: allow disable color vars, radius, global styles

### LINKED ISSUE #3: Add support for custom theming with css variables in a css file
### Clear and concise description of the problem

Hello there,

First of all i want to thank you for this, its a game changer for someone like me who loves both unocss and shadcn/ui.

I would love to propose for more flexibility in the usage of this preset. When using shadcn/ui with tailwind, you get the flexibility of customizing your theme colors and values using the css variables provided to you through the tailwindcss file which is undoubtedly very useful.

I wonder if we could also have this preset give room for such customization.

### Suggested solution

Here are some possible solutions.

## 1. Passing custom variable values to the preset

```js
presetShadcn({
  theme: {
        default: {
          background: "0 0% 100%",
          // and so on....
        },
        dark: {
          background: "0 0% 100%",
          // and so on....
        }
        cherry: {
          background: "0 0% 100%",
          // and so on....
        }
   }
})
```

## 2. Switching to custom theme
We could have some sort of property that lets the preset know that we want to use custom theme values.
This will then let you set your values in a css file, without loading the inbuilt ones
```ts
presetShadcn({
   customThemeVars: true
})
```

```css
::root{
  ---background: 255 255% 100%;
 // and so on....
}


.dark{
  ---background: 0 0% 100%;
 // and so on....
}
```

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/hyoban/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/hyoban/.github/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by NamesMT**
### Description
This would allow user to config the color variables in css file, as well as finer control over the global styles behavior

### Linked Issues
Related / Resolves #3

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export type ThemeOptions = {
   /**
    * @default 'zinc'
    */
-  color?: ColorOptions
+  color?: ColorOptions | null
```
Comment (hyoban) [inline src/types.ts:21]: Maybe use `false`?

File: src/generate.ts [hunk 2]
```diff
@@ -36,16 +57,16 @@ function getColorTheme(color: ColorOptions) {
 
   if (typeof color === 'string') {
     name = color
-    ;({ light, dark } = getBuiltInTheme(color))
+    ; ({ light, dark } = getBuiltInTheme(color))
```
Comment (hyoban) [inline src/generate.ts:60]: Can you help me remove these formatting changes?

File: src/types.ts (hunk 1 above)
Comment (NamesMT) [reply src/types.ts:21]: Sure, I think I'll also add false for radius?

File: src/generate.ts (hunk 2 above)
Comment (NamesMT) [reply src/generate.ts:60]: Okay, that wasn't expected haha, might be a bit sleepy yesterday

File: src/types.ts (hunk 1 above)
Comment (hyoban) [reply src/types.ts]: Yes, It's OK

**Comment by NamesMT** [comment]
Hi @hyoban, I just force pushed the PR with the following changes:
+ Change `null` to `false`
+ Also allowing to disable `radius` with `false` when in combination with `color`
+ DRYing `colorCSSVarsStyles`'s return
+ Rebased to `main` to resolve conflict

File: src/index.ts [hunk 3]
```diff
@@ -21,14 +25,7 @@ export function presetShadcn(options: PresetShadcnOptions = {}): Preset<Theme> {
...
 
-          * {
-            border-color: hsl(var(--border));
-          }
-
-          body {
-            color: hsl(var(--foreground));
-            background: hsl(var(--background));
-          }
+          ${globals ? generateGlobalStyles() : ''}
```
Comment (hyoban) [inline src/index.ts:28]: I think maybe we don't need this option. It’s always good to have this.

Comment (NamesMT) [reply src/index.ts:28]: Not sure about that, it's always better to allow more control, yesterday I was setting up `PrimeVue tailwind preset mode` + `Shadcn` using UnoCSS, and I was wondering what overridden my background color as I did set it in my `main.(s)css` file.
UnoCSS styles (`__uno.css`) is generated after Nuxt's `css` config and could introduce some priority headaches for user control.

Some image for reference:
image
image

File: src/index.ts [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
 
 export const builtinColors = themes.map(theme => theme.name)
 export const builtinRadiuses = [0, 0.3, 0.5, 0.75, 1] as const
 
-export function presetShadcn(options: PresetShadcnOptions = {}): Preset<Theme> {
+/**
+ * @param globals Generates global variables, like *.border-color, body.color, body.background.
+ * @default true
+ */
+export function presetShadcn(options: PresetShadcnOptions = {}, globals?: boolean): Preset<Theme> {
```
Comment (hyoban) [inline src/index.ts:15]: Missing default option?

**Comment by NamesMT** [comment]
@hyoban I have added the missing default for globals and added tests.
I also moved test snapshots to a sub directory and ignore them in eslint, is that ok?

**Comment by hyoban** [comment]
LGTM, Thank you so much for your contribution.

**Comment by hyoban** [comment]
Released at 0.3.0, you can give it a try.

**Pull request merged**